### ISSUE TITLE
Cannot compile "verifier->cb.cb = verify_certificate_callback" : incompatible pointer type 

### ISSUE THREAD

**Issue by bortzmeyer** [private-octopus/picoquic#1193]
With the current HEAD:

```
% make
[  1%] Building C object CMakeFiles/picoquic-core.dir/picoquic/tls_api.c.o
/home/stephane/IETF/Hackathon-107/DoQ/picoquic/picoquic-git/picoquic/tls_api.c: In function ‘picoquic_enable_custom_verify_certificate_callback’:
/home/stephane/IETF/Hackathon-107/DoQ/picoquic/picoquic-git/picoquic/tls_api.c:1266:25: error: assignment to ‘int (*)(struct st_ptls_verify_certificate_t *, ptls_t *, int (**)(void *, uint16_t,  ptls_iovec_t,  ptls_iovec_t), void **, ptls_iovec_t *, size_t)’ {aka ‘int (*)(struct st_ptls_verify_certificate_t *, struct st_ptls_t *, int (**)(void *, short unsigned int,  struct st_ptls_iovec_t,  struct st_ptls_iovec_t), void **, struct st_ptls_iovec_t *, long unsigned int)’} from incompatible pointer type ‘int (*)(ptls_verify_certificate_t *, ptls_t *, int (**)(void *, ptls_iovec_t,  ptls_iovec_t), void **, ptls_iovec_t *, size_t)’ {aka ‘int (*)(struct st_ptls_verify_certificate_t *, struct st_ptls_t *, int (**)(void *, struct st_ptls_iovec_t,  struct st_ptls_iovec_t), void **, struct st_ptls_iovec_t *, long unsigned int)’} [-Werror=incompatible-pointer-types]
 1266 |         verifier->cb.cb = verify_certificate_callback;
      |                         ^
cc1: all warnings being treated as errors
make[2]: *** [CMakeFiles/picoquic-core.dir/build.make:368: CMakeFiles/picoquic-core.dir/picoquic/tls_api.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/picoquic-core.dir/all] Error 2
make: *** [Makefile:149: all] Error 2

% gcc --version
gcc (Debian 10.2.1-6) 10.2.1 20210110
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

After casting verify_certificate_callback to the proper type, it compiles and seems to run.

**Comment by victorstewart** [comment]
that line no longer exists in the master?

see:
[https://github.com/private-octopus/picoquic/blob/master/picoquic/tls_api.c](https://github.com/private-octopus/picoquic/blob/master/picoquic/tls_api.c)

there was a recent change in the api on the picotls side. so make sure you're running the newest versions of both.

**Comment by huitema** [comment]
You need to update to the latest version of picotls. There was an incompatible change in that library.

-- Christian Huitema

**Comment by huitema** [comment]
@bortzmeyer this is a duplicate of issue #1185, which was fixed in PR #1186. It happened because you refreshed your copy of picotls and picked the API change, but did not yet update your copy of picoquic to the most recent version.

**Comment by bortzmeyer** [comment]
Actually, it was even more stupid than that, I was on the wrong branch. Sorry for the false alarm.
